vcx/libvcx/src/connection.rs

```diff
@@ -96,6 +97,99 @@ struct Connection {
     version: Option<settings::ProtocolTypes>
 }
 
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+pub struct MyAgentInfo {
```

Comment:
Do we really need this complexity of getting `AgentInfo`.
Why can't we just add a single function that returns the whole structure at once?

Reply:
There are two sets of information which are known at different phases.
1. Information known after provision: (agency_did, agency_vk, agent_did, agent_vk) 
2. The rest is information known after a pw connection. 

There are time where someone may want the agent information known only after provisioning and another where they want the entire set after a connection is established.